modular_zubbers/code/modules/antagonists/bloodsucker/bloodsuckers/bloodsucker.dm

```diff
@@ -164,7 +159,8 @@
 		QDEL_NULL(vamprank_display)
 
 	SSsunlight.remove_sun_sufferer(owner.current) //check if sunlight should end
-	current_mob.dna?.species?.on_bloodsucker_loss(current_mob)
+	if(iscarbon(current_mob))
+		current_mob.dna.species?.on_bloodsucker_loss(current_mob)
```

Comment:
I don't think species needs to be nullchecked here as it should literally always exist if dna exists.

However dna should be nullchecked since that's null on some carbons